pkg/api/getambassador.io/v3alpha1/crd_listener.go

```diff
@@ -132,6 +132,13 @@ type HostBindingType struct {
 	Selector  *metav1.LabelSelector `json:"selector,omitempty"`
 }
 
+//HTTP3Options provides options to set when a Listener is enabled to understand HTTP/3
+type HTTP3Options struct {
+	//AltSvc allows modifying the default `alt-svc` header that is attached to all HTTP/1 & HTTP/2 response indicating HTTP/3 support.
+	// +kubebuilder:default="h3=\":443\"; ma=86400, h3-29=\":443\"; ma=86400"
+	AltSvc string `json:"altSvc,omitempty"`
+}
```

Comment:
1. I don't like exposing this as a raw string. This feels like something where we really want to "own". That we want the user to just tell us what port H/3 is on, and maybe max age. That we should take on the "expert-knowledge" of forming that in to a valid `Alt-Svc` header. And The user really shouldn't be configuring the list of protocol IDs themselves; what's correct there just depends on what our Envoy supports, and so that's something we should be responsible for tracking, not shifting that on to the user.

2. If we are going to add a raw-string like this, we should probably have the doc comment include a link to the relevant spec or docs to understand the semantics of that header.

3. style nits: please add a space between `//` and the start of the text, and please add a blank `//` line between the main text and the `+kubebuilder` line.

Reply:
valid feedback!

This has been removed per our discussion offline and future enhancements will take into account your feedback.